src/main/java/org/jenkinsci/plugins/authorizeproject/strategy/SpecificUsersAuthorizationStrategy.java

```diff
@@ -117,51 +172,15 @@ public Authentication authenticate(Job<?, ?> project, Queue.Item item) {
             return Jenkins.ANONYMOUS;
         }
     }
-    
+
     /**
-     * Returns whether authentication is required to update the configuration to newStrategy.
-     * 
-     * @param newStrategy strategy to be configured.
-     * @param currentStrategy strategy now configured.
-     * @return whether authentication is required.
+     * {@inheritDoc}
      */
-    protected static boolean isAuthenticateionRequired(
-            SpecificUsersAuthorizationStrategy newStrategy,
-            SpecificUsersAuthorizationStrategy currentStrategy
-    ) {
-        if (newStrategy == null) {
-            // if configure is removed, no need to authenticate.
-            return false;
-        }
-        
-        if (Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER)) {
-            // Administrator can specify any user.
-            return false;
-        }
-        
-        User u = User.current();
-        if (u != null && AuthorizeProjectUtil.userIdEquals(u.getId(), newStrategy.getUserid())) {
-            // Any user can specify oneself.
-            return false;
-        }
-        
-        if (currentStrategy == null) {
-            // if currentStrategy is null, authentication is always required.
-            return true;
-        }
-
-        if (
-                currentStrategy.isNoNeedReauthentication()
-                && AuthorizeProjectUtil.userIdEquals(currentStrategy.getUserid(), newStrategy.getUserid())
-        ) {
-            // the specified user is not changed, 
-            // and specified that authentication is not required in that case.
-            return false;
-        }
-        
-        return true;
+    @Override
+    public boolean hasConfigurePermission(AccessControlled context) {
+        return AuthorizeProjectUtil.userIdEquals(Jenkins.getAuthentication().getName(), userid);

```

Comment:
Also should accept users with `ADMINISTER`.
And administrators always has permission to configure, it should make sense to test by the caller (`AuthorizeProjectProperty#readResolve`).